client/src/Main.purs

```diff
@@ -132,31 +133,32 @@ compile = do
   displayLoadingMessage
   clearAnnotations
 
-  runContT (runExceptT (API.compile Config.compileUrl code)) \res_ ->
+  launchAff_ $ runExceptT (API.compile Config.compileUrl code) >>= \res_ ->
     case res_ of
-      Left err -> displayPlainText err
+      Left err -> liftEffect $ displayPlainText err
       Right res -> do
-        cleanUpMarkers
+        liftEffect cleanUpMarkers
 
         case res of
           Right (CompileSuccess (SuccessResult { js, warnings })) -> do
-            showJs <- isShowJsChecked
-            if showJs
-              then do hideLoadingMessage
-                      displayPlainText js
-              else runContT (runExceptT $ runLoader loader (JS js)) \sources -> do
-                     hideLoadingMessage
-                     for_ warnings \warnings_ -> do
-                       let toAnnotation (CompileWarning{ errorCode, position, message }) =
-                             position <#> \(ErrorPosition pos) ->
-                               { row: pos.startLine - 1
-                               , column: pos.startColumn - 1
-                               , type: "warning"
-                               , text: message
-                               }
-                       runEffectFn1 setAnnotations (mapMaybe toAnnotation warnings_)
-                     for_ sources (execute (JS js))
-          Right (CompileFailed (FailedResult { error })) -> do
+            showJs <- liftEffect isShowJsChecked
```

Comment:
These changes are largely indentation related; the only real difference is the introduction of `liftEffect` and switching `runContT` for binds in `Aff`.